packages/-ember-data/tests/integration/identifiers/scenarios-test.ts

```diff
@@ -0,0 +1,678 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import { IDENTIFIERS } from '@ember-data/canary-features';
+import Store, {
+  recordIdentifierFor,
+  setIdentifierUpdateMethod,
+  setIdentifierGenerationMethod,
+  setIdentifierResetMethod,
+  setIdentifierForgetMethod,
+} from '@ember-data/store';
+import Model, { attr } from '@ember-data/model';
+import Adapter from '@ember-data/adapter';
+import Serializer from '@ember-data/serializer';
+import { resolve, all } from 'rsvp';
+import { ExistingResourceObject } from '@ember-data/store/-private/ts-interfaces/ember-data-json-api';
+import { Dict } from '@ember-data/store/-private/ts-interfaces/utils';
+import { StableRecordIdentifier } from '@ember-data/store/-private/ts-interfaces/identifier';
+import { identifierCacheFor } from '@ember-data/store/-private';
+import { set } from '@ember/object';
+
+function isNonEmptyString(str: any): str is string {
+  return typeof str === 'string' && str.length > 0;
+}
+
+if (IDENTIFIERS) {
+  module('Integration | Identifiers - scenarios', function(hooks) {
+    setupTest(hooks);
+
+    module('Secondary Cache based on an attribute', function(hooks) {
+      let store;
+      let calls;
+      let secondaryCache: {
+        id: Dict<string, string>;
```

Comment:
When did the `Dict` type get changed to allow two typeparams? (`{ [KK in K]: V; }`). It's now not really a dictionary (arbitrary keys) and matches nearly exactly with the built-in `Record<K, V>` utility type.

Reply:
Great catch for two reasons:

1) I was not aware of `Record<K, V>` being a built-in type and that explains why I've sometimes found that definition trying to trump our own defined `Record` type.

2) As far as I can recall this was the `Dict` we introduced together a while back.  I'll make a cleanup pass to refactor to `Dict<ValueType>`

Reply:
(said cleanup pass will be separate from this PR as `Dict` has this <K, V>` usage more broadly)